main/src/ca/uwaterloo/flix/language/phase/typer/TypeReduction.scala

```diff
@@ -131,37 +130,40 @@ object TypeReduction {
    * @param loc     the location where the java method has been called
    * @return        A ResolutionResult object that indicates the status of the resolution progress
    */
-  def lookupMethod(thisObj: Type, method: String, ts: List[Type], loc: SourceLocation)(implicit flix: Flix): ResolutionResult2 =
+  def lookupMethod(thisObj: Type, method: String, ts: List[Type], loc: SourceLocation)(implicit flix: Flix): JavaResolutionResult =
     thisObj match {
-      case Type.Cst(TypeConstructor.Str, loc2) =>
+      case Type.Cst(TypeConstructor.Str, _) =>
         val clazz = classOf[String]
         // Filter out candidate methods by method name
         // NB: this considers also static methods
+        val nbParams = ts.length - ts.foldLeft(0) { (acc, t) =>
+          t match {
+            case Type.Cst(TypeConstructor.Unit, _) => 1 + acc
```

Comment:
It looks like we are getting rid of Units here, which is the same work we did in the Weeder. Do we have to do it twice?

Reply:
Indeed it should now work without since the parser does it. I removed it and it seems to be working fine.